devbuild.sh

```diff
@@ -1,76 +1,235 @@
 #!/bin/bash
-set -eu
-
-#cd to location of script
-MYDIR=$(cd "$(dirname "$(readlink -f -n "${BASH_SOURCE[0]}" )" )" && pwd -P)
 
+# usage instructions
 usage () {
-  echo "Usage: "
-  echo "  $0 PLATFORM COMPILER"
-  echo ""
-  echo "PLATFORM: Name of machine you are building on"
-  echo "COMPILER: (optional) compiler to use; valid options are 'intel', 'gnu'"
-  echo ""
-  echo "NOTE: This script is for internal developer use only;"
-  echo "See User's Guide for detailed build instructions"
+  printf "Usage: $0 [OPTIONS]...\n"
+  printf "\n"
+  printf "OPTIONS\n"
+  printf "  -h, --help\n"
+  printf "      show this help guide\n"
+  printf "  --platform=PLATFORM\n"
+  printf "      name of machine you are building on; defauts to hostname\n"
+  printf "      (.e.g. cheyenne | hera | jet | orion | wcoss)\n"
+  printf "  --compiler=COMPILER\n"
+  printf "      compiler to use; default depends on platform\n"
+  printf "      (.e.g. intel | gnu | cray | gccgfortran)\n"
+  printf "  --app=APPLICATION\n"
+  printf "      weather model application to build\n"
+  printf "      (.e.g. ATM | ATMW | S2S | S2SW)\n"
+  printf "  --ccpp=\"CCPP_SUITE1,CCPP_SUITE2...\"\n"
+  printf "      CCCP suites to include in build; delimited with ','\n"
+  printf "  --components=\"COMPONENT1,COMPONENT2...\"\n"
+  printf "      components to include in build; delimited with ','\n"
+  printf "  --continue\n"
+  printf "      continue with existing build\n"
+  printf "  --clean\n"
+  printf "      removes existing build; overrides --continue\n"
+  printf "  --build-dir=BUILD_DIR\n"
+  printf "      build directory\n"
+  printf "  --install-dir=INSTALL_DIR\n"
+  printf "      installation prefix\n"
+  printf "  --build-type=BUILD_TYPE\n"
+  printf "      build type; defaults to RELEASE\n"
+  printf "      (.e.g. DEBUG | RELEASE | RELWITHDEBINFO)\n"
+  printf "  --build-jobs=BUILD_JOBS\n"
+  printf "      number of build jobs; defaults to 4\n"
+  printf "  -v, --verbose\n"
+  printf "      build with verbose output\n"
+  printf "\n"
+  printf "NOTE: This script is for internal developer use only;\n"
+  printf "See User's Guide for detailed build instructions\n"
 }
 
-PLATFORM="${1:-NONE}"
-COMPILER="${2:-intel}"
+# print settings
+settings () {
+  printf "Settings:\n"
+  printf "\n"
+  printf "  SRC_DIR=${SRC_DIR}\n"
+  printf "  BUILD_DIR=${BUILD_DIR}\n"
+  printf "  INSTALL_DIR=${INSTALL_DIR}\n"
+  printf "  PLATFORM=${PLATFORM}\n"
+  printf "  COMPILER=${COMPILER}\n"
+  if [ ! -z "${APPLICATION}" ]; then printf "  APP=${APPLICATION}\n"; fi
+  if [ ! -z "${CCPP}" ]; then printf "  CCPP=${CCPP}\n"; fi
+  if [ ! -z "${COMPONENTS}" ]; then printf "  COMPONENTS=${COMPONENTS}\n"; fi
+  printf "  CLEAN=${CLEAN}\n"
+  printf "  CONTINUE=${CONTINUE}\n"
+  printf "  BUILD_TYPE=${BUILD_TYPE}\n"
+  printf "  BUILD_JOBS=${BUILD_JOBS}\n"
+  printf "  VERBOSE=${VERBOSE}\n"
+  printf "\n"
+}
 
+# find system name
+find_system () {
```

Comment:
There is already a script (from EMC, I think) in the UFS_UTILS repo that does this.  It's at:
`ufs-srweather-app/src/UFS_UTILS/sorc/machine-setup.sh
`
Maybe you can run that here to avoid code duplication.

Reply:
My concern is that src/UFS_UTILS is an external that is dependent on using manage externals. I can write this subroutine to look for machine-setup.sh and fail if it doesn't find it or try to determine the system name on its own if it doesn't find it. What's your preference?

Reply:
The second approach sounds good to me -- look for machine-setup.sh, otherwise try to determine the machine on its own.

Reply:
I'm now using machine-setup.sh unless the file is not found.